### DISCUSSION TITLE
Caused by: java.sql.SQLException: No suitable driver

### DISCUSSION THREAD

**Discussion by rnei00** [aws/aws-advanced-jdbc-wrapper#668] [General]
Hello,
Some issue toconnect to my PostgreSQL database : 
`Caused by: java.sql.SQLException: No suitable driver
	at java.sql/java.sql.DriverManager.getDriver(Unknown Source)
	at software.amazon.jdbc.Driver.connect(Driver.java:97)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql/java.sql.DriverManager.getConnection(Unknown Source)
	at be.totalenergies.ctrm.integration.shipadapter.lambda.RetentionHandler.getDatabaseConnection(RetentionHandler.java:71)
	at be.totalenergies.ctrm.integration.shipadapter.lambda.RetentionHandler.handleRequest(RetentionHandler.java:41)
	... 4 more
`

Code : 

```
try {
            Class.forName("software.amazon.jdbc.Driver").getDeclaredConstructor().newInstance();
        } catch (ClassNotFoundException e) {
            System.err.println("JDBC driver not found!");
        }
        String connectionString = "jdbc:aws-wrapper:postgresql://" + host + ":" + port + "/" + name;
        Properties properties = new Properties();
        properties.setProperty("driver", "software.amazon.jdbc.Driver");
        properties.setProperty(PropertyDefinition.PLUGINS.name, "iam");
        properties.setProperty(PropertyDefinition.USER.name, user);
       return DriverManager.getConnection(connectionString, properties);
```

Someone can help me :p ?

**Comment by karenc-bq** [comment]
Hi @rnei00, thank you for reaching out.

- Can you please ensure you have the PostgreSQL community driver registered in the classpath? Please see [this section](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/GettingStarted.md#minimum-requirements) for more information.
- Since you are using the IAM plugin, could you please also ensure you have the [AWS SDK for Java](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/docs/using-the-jdbc-driver/using-plugins/UsingTheIamAuthenticationPlugin.md#prerequisites) in your classpath?
- [Here is a simple example](https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/examples/AWSDriverExample/src/main/java/software/amazon/AwsIamAuthenticationPostgresqlExample.java) for using the AWS JDBC Driver with the IAM plugin, note that you don't have to set the `driver` property.

**Comment by rnei00** [comment]
I need to add "Class.forName("org.postgresql.Driver"); " without it's not worked

```
Class.forName("org.postgresql.Driver");
        Properties properties = new Properties();
        properties.setProperty(PropertyDefinition.PLUGINS.name, "iam");
        properties.setProperty(PropertyDefinition.USER.name, user);

        String connectionString = "jdbc:aws-wrapper:postgresql://" + host + ":" + port + "/" + name;

        return DriverManager.getConnection(connectionString, properties);

```
